pkg/apiserver/debugapi/endpoint/param.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import "net/url"
+
+type APIParamModel interface {
+	AddMiddleware(handler ...ModelMiddlewareHandlerFunc) APIParamModel
+	GetMiddlewares(param *APIParam, isPathParam bool) []MiddlewareHandler
+}
```

Comment:
This looks very similar to `Middlewarer` but not exactly the same. Is it a mistake?

Reply:
I want to make it easier to define the model middleware and restrict data access to the model middleware, so the interface is different. A little strange, any better suggestions?